### PR TITLE
Add monitoring

### PR DESCRIPTION
**Pull request by simonwheatley**

### CODE REVIEW

**Comment by joshbetz** [comment]
`WPCOM_VIP_Query_Log` seems like a lot of overhead for what comes down to essentially 1 error_log statement on shutdown. Getting rid of the singleton would reduce it to ~10 lines of code.

**Comment by simonwheatley** [comment]
@joshbetz Fair point, I thought I was going to have to store the queries somewhere when I started… but then I remembered `WPDB::queries`.

File: misc.php [hunk 1]
```diff
@@ -140,3 +140,21 @@ function _vip_filter_rest_url_for_ssl( $url ) {
 
 	return $url;
 }
+
+
+function wpcom_vip_query_log() {
+	if ( '/cache-healthcheck?' === $_SERVER['REQUEST_URI'] ) {
+		return;
+	}
+	$num_queries = count( $GLOBALS['wpdb']->queries );
```
Comment (joshbetz) [inline misc.php]: I think `$wpdb->num_rows` will do the same thing?

**Comment by joshbetz** [comment]
Looks good to me.

**Comment by simonwheatley** [comment]
I'll probably remove the Jetpack `updating_jetpack_version` before deploying it to production…

**Comment by simonwheatley** [comment]
I've added some conditional checks for constants, so we can carefully control which sites get instrumented and we don't end up running lots of logging everywhere unexpectedly.

File: misc.php (hunk 1 above)
Comment (nickdaugherty) [inline misc.php]: This will be empty unless SAVEQUERIES has been set, right?

Comment (simonwheatley) [inline misc.php]: Fair point, I'll conditionally set `SAVEQUERIES`.

Comment (simonwheatley) [inline misc.php]: Fair point, I'll conditionally set `SAVEQUERIES`.

**Pull request merged**